client/homescreen/layout.js

```diff
@@ -185,7 +220,10 @@ export default compose(
 		);
 
 		const shouldShowWelcomeModal =
-			welcomeModalDismissedHasResolved && ! welcomeModalDismissed;
+			welcomeModalDismissedHasResolved &&
+			! welcomeModalDismissed &&
+			welcomeFromCalypsoModalDismissedResolved &&
+			! welcomeFromCalypsoModalDismissed;
```

Comment:
Should we change the last item here to `! shouldShowWelcomeFromCalypsoModal` instead of `! welcomeFromCalypsoModalDismissed`.
I don't know how common the case is that the user visits Woo admin for the first time coming from calypso, but this would trigger both `shouldShowWelcomeModal` and `shouldShowWelcomeFromCalypsoModal` to be true at the same time.

Actually causing a strange effect, where the welcome modal gets quickly loaded and closed, and then showing the calypso modal, as you can see from the tracks that it generated (on initial window load).
Triggering the open track for both modal's, but then immediately closing the welcome modal. 
<img width="1411" alt="Screen Shot 2021-01-06 at 9 24 05 AM" src="https://user-images.githubusercontent.com/2240960/103773900-240e3680-5002-11eb-8b73-3284056efda7.png">

@becdetat would it be worth in this case showing the welcome modal, but replacing the first page with the Calypso page?

Reply:
> Code looks great! I just left one comment for a particular use case where the user visits the WooCommerce store for the first time from Calypso, causing some weird things behind the scenes, see comment below.

cc @pmcpinto to see if this use case might actually happen.

Reply:
@becdetat @louwie17 if he's a brand new user he shouldn't see the modal because he never interacted with the Store UI. Makes sense? Or maybe I'm missing something here.

cc @elizaan36 for additional feedback

Reply:
> if he's a brand new user he shouldn't see the modal because he never interacted with the Store UI.

@pmcpinto which modal are you referring to here that shouldn't be shown? The calypso welcome modal or the original welcome modal (the one with 3 pages, mentioning the inbox, and performance stats).

I didn't test this with Calypso, but the way the logic is set up, is that a brand new user would only see the single page modal - Calypso welcome screen (new store management experience).

Reply:
@louwie17 
> the way the logic is set up, is that a brand new user would only see the single page modal - Calypso welcome screen (new store management experience)

that was certainly my intent as well. A new user from Calypso should only see the new Calypso welcome modal, and never the original welcome modal. This corresponds with @elizaan36 's design for the new Calypso welcome modal as well.

Reply:
Just to clarify, the Calypso welcome modal would only be relevant for the existing Calypso users, so anyone that's been a Store user or becomes a Store user during the sunset period. 

If we're talking about brand-new Store users *after* Store sunset, then they should see the regular multi-page welcome modal upon landing on the Home screen. That may be the logic we have, but just wanted to clarify.

Reply:
@louwie17 I'm mainly talking about what Elizabeth mentioned here:
 
> If we're talking about brand-new Store users after Store sunset, then they should see the regular multi-page welcome modal upon landing on the Home screen. That may be the logic we have, but just wanted to clarify.

If a user signs up during the sunset period, they should also see the regular welcome modal. Makes sense?

Reply:
Great, thanks for the clarification everyone. I believe everything is working as expected then.
Approving this PR.